Check hyperlinks for facebook links

changhuapeng opened the pull request:
I have added a module to check for links that are not allowed to be post. It only checks for facebook link at the moment (not allowed on r/singapore) but new domains can be added to the list in ~~links.json~~ settings.py like this:

    BLOCKED_LINKS = [
        "facebook.com",
        "example.com"
    ]

changhuapeng committed:
Check hyperlinks for facebook links

fterh commented:
Hey, so sorry, I've rewritten a lot of the code since then. If you can update your code to be compatible with v4.0 I'll be happy to merge it!

Thanks! :)

Commit:
Merge branch 'master' into disallowed_link

changhuapeng committed:
Clean and merge codes with v4.0

changhuapeng commented:
I have updated and merged the codes to be compatible with v4.0. With this update, I have updated certain variables and class names to follow the python style guide, removed article.py which is not used at all, and reduced redundant "read from file" operations by moving both the lists of "approved"  sites / "blocked" links from sites.txt and links.json into settings.py

fterh commented:
Thanks so much, I'll have a look at it ASAP. Currently, I'm facing a bug where the script is running fine (no errors/crashes) but it's not posting anything.

changhuapeng commented on modules/core.py:
```diff
@@ -1,8 +1,8 @@
-from modules.mercury import mercury
+from modules.mercury import Mercury
 
 def core(submission, replied, settings, mention=False, summoned=False):
```
~~is `mention=False` supposed to be `mention=None`?~~

Edit: I've just saw this change that you have made in the latest bugfix commit.

fterh commented:
Hey, I see that you hardcoded `APPROVED_SITES` and `BLOCKED_LINKS` to settings.py. I'd prefer to have these on external files, so that they can be modified without having to dive into the code.

Also, my telegram handle is "fabian1994", feel free to PM me for faster response!

changhuapeng committed:
Read list of blocked/approved sites from external json file

changhuapeng committed:
Merge branch 'master' into disallowed_link

fterh commented:
Hey, I'm not sure if you see this line but from my side it says: This branch has conflicts that must be resolved. Could you resolve them on your end? (I'm worried about breaking your code).

Also, I'll be traveling in the US the next couple weeks so I might not be as responsive. If you can confirm with reasonable confidence that there are no major bugs, I'm happy to deploy it once the conflicts are resolved :)

changhuapeng committed:
Merge branch 'master' into disallowed_link

changhuapeng commented:
Not too sure why the conflicts weren't resolved yesterday after merging the branches but I have just resolved that.

fterh commented:
I've merged it, I'll SSH into my AWS instance and `git pull` shortly. LMK if it doesn't work so I can revert!
